python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -407,64 +409,51 @@ class RandomForestClassifier(Base):
-        with cython.boundscheck(False):
-            model_pbuf_vec.assign(& model_pbuf_mv[0],
-                                  & model_pbuf_mv[model_pbuf_mv.shape[0]])
-
-        task_category = CLASSIFICATION_MODEL
-        build_treelite_forest(
-            & cuml_model_ptr,
-            rf_forest,
-            <int> self.n_cols,
-            <int> task_category,
-            model_pbuf_vec)
-        mod_ptr = <uintptr_t> cuml_model_ptr
-        self.treelite_handle = ctypes.c_void_p(mod_ptr).value
+        if self.model_bytes:  # bytes -> Treelite
+            tl_handle_int = treelite_deserialize(self.model_bytes)
+            tl_handle = <ModelHandle>tl_handle_int
+        else:                 # RF -> Treelite
+            task_category = CLASSIFICATION_MODEL
+            build_treelite_forest(
+                &tl_handle,
+                rf_forest,
+                <int> self.n_cols,
+                <int> task_category)
+        self.treelite_handle = ctypes.c_void_p(<uintptr_t>tl_handle).value
         return self.treelite_handle
 
-    def _get_protobuf_bytes(self):
+    def _get_model_bytes(self):
         """
-        Returns the self.model_pbuf_bytes.
+        Returns the self.model_bytes.
         Cuml RF model gets converted to treelite protobuf bytes by:
             1. converting the cuml RF model to a treelite model. The treelite
             models handle (pointer) is returned
-            2. The treelite model handle is used to convert the treelite model
-            to a treelite protobuf model which is stored in a temporary file.
-            The protobuf model information is read from the temporary file and
-            the byte information is returned.
-        The treelite handle is stored `self.treelite_handle` and the treelite
-        protobuf model bytes are stored in `self.model_pbuf_bytes`. If either
-        of information is already present in the model then the respective
-        step is skipped.
+            2. The treelite model handle is converted to bytes.
+        The treelite model bytes are stored in `self.model_bytes`. If the
+        model bytes are present, we can skip _obtain_treelite_handle().
         """
-        if self.model_pbuf_bytes:
-            return self.model_pbuf_bytes
+        if self.model_bytes:
+            return self.model_bytes
         elif self.treelite_handle:
             fit_mod_ptr = self.treelite_handle
         else:
             fit_mod_ptr = self._obtain_treelite_handle()
         cdef uintptr_t model_ptr = <uintptr_t> fit_mod_ptr
-        cdef vector[unsigned char] pbuf_mod_info = \
-            save_model(<ModelHandle> model_ptr)
-        cdef unsigned char[::1] pbuf_mod_view = \
-            <unsigned char[:pbuf_mod_info.size():1]>pbuf_mod_info.data()
-        self.model_pbuf_bytes = bytearray(memoryview(pbuf_mod_view))
-        return self.model_pbuf_bytes
+        self.model_bytes = treelite_serialize(model_ptr)
```

Comment:
Given this is now a Python `dict`, maybe we should rename `model_bytes` to something else more descriptive (`model_serialized` perhaps?).

Reply:
Done.